### PR TITLE
corrected the destination path for bash completion files in Makefile and *.spec file

### PR DESCRIPTION
**Pull request by pseiler**
"/etc/bash_completion.d/" is used for user deployed files
$(prefix)/bash-completion/completions/ is the correct path for system/rpm deployed files
Check output from rpm -ql for more details
```bash
rpm -ql bash-completion
```

I found this issue when packaging the subscription-manager for SLExx and this is useful in general.

### CODE REVIEW

**Comment by cnsnyder** [comment]
@candlepin-pull-request-bot ok to test

**Comment by pseiler** [comment]
@cnsnyder I did a fatal mistake. I used the wrong macro. Fixed it in the last commit

**Comment by pseiler** [comment]
Fixed the missing change in the last commit. Now both files should work together

**Comment by pseiler** [comment]
@cnsnyder anything left to fix in this pull request. I prepared some more if this one gets accepted

**Comment by wottop** [comment]
Please squash the changes into a single commit.

**Comment by wottop** [comment]
buildbot: retest this please

File: Makefile [hunk 1]
```diff
@@ -153,7 +153,7 @@ dbus-common-install:
 	install -d $(DESTDIR)/etc/dbus-1/system.d
 	install -d $(DESTDIR)/$(INSTALL_DIR)/dbus-1/system-services
 	install -d $(DESTDIR)/$(LIBEXEC_DIR)
-	install -d $(DESTDIR)/etc/bash_completion.d
```
Comment (cnsnyder) [inline Makefile]: At least for SLES11 support we need to install to /etc/bash_completion.d. I'd suggest we conditionally do that.

Comment (pseiler) [reply Makefile]: You're right. I'll work on a solution, and squash my commits. You hear from me.
Thank you ;)

**Comment by pseiler** [comment]
@cnsnyder  Please have a look at @awood 's commit. This looks good to me.
Sorry for this much delay!

**Comment by awood** [comment]
@candlepin-pull-request-bot retest this please

**Comment by awood** [comment]
@cnsnyder any idea what these failures are about?  They seem unrelated:

```
CalledProcessError: Command 'test "$(find /var/cache/zypp/packages/ -name 'sos*.rpm' | wc -l)" -gt 0' returned non-zero exit status 1
```

**Comment by pseiler** [comment]
@awood I think this checks if some sos package was downloaded?
Imo it's not related for SUSE packages, since sos* packages are only for redhat based distributions.

**Comment by nikosmoum** [comment]
@awood the suse/zypper failures are unrelated to this PR and have been failing consistently for a while. I managed to fix another one but this one persists..

**Comment by awood** [comment]
@candlepin-pull-request-bot retest this please

@pseiler I had to fix a conflict.  Once we get a (semi) passing run of the CI environment, I'll merge.

**Pull request merged**